packages/accounts/src/msca/multi-owner-account.ts

```diff
@@ -0,0 +1,124 @@
+import {
+  SignerSchema,
+  createBaseSmartAccountParamsSchema,
+  type SignTypedDataParams,
+  type SupportedTransports,
+} from "@alchemy/aa-core";
+import {
+  concatHex,
+  encodeFunctionData,
+  hashMessage,
+  hashTypedData,
+  hexToBytes,
+  isBytes,
+  toBytes,
+  type FallbackTransport,
+  type Hash,
+  type Transport,
+} from "viem";
+import { z } from "zod";
+import { MultiOwnerMSCAFactoryAbi } from "./abis/MultiOwnerMSCAFactory.js";
+import { MSCABuilder, StandardExecutor } from "./builder.js";
+import { MultiOwnerPlugin } from "./plugins/multi-owner.js";
+
+export const createMultiOwnerMSCASchema = <
+  TTransport extends SupportedTransports = Transport
+>() =>
+  createBaseSmartAccountParamsSchema<TTransport>().extend({
+    owner: SignerSchema,
+    index: z.bigint().optional().default(0n),
+  });
+
+export type MultiOwnerMSCAParams = z.input<
+  ReturnType<typeof createMultiOwnerMSCASchema>
+>;
+
+export const createMultiOwnerMSCABuilder = <
+  TTransport extends Transport | FallbackTransport = Transport
+>(
+  params_: MultiOwnerMSCAParams
+) => {
+  const params = createMultiOwnerMSCASchema<TTransport>().parse(params_);
+
+  const builder = new MSCABuilder()
+    .withFactory(async (acct) => {
+      const ownerAddress = await params.owner.getAddress();
+      return concatHex([
+        acct.getFactoryAddress(),
+        encodeFunctionData({
+          abi: MultiOwnerMSCAFactoryAbi,
+          functionName: "createAccount",
+          // TODO: this needs to support creating accounts with multiple owners
+          args: [params.index, [ownerAddress]],
+        }),
+      ]);
+    })
```

Comment:
so `MultiOwnerMSCAFactory` is always the factory that create MSCAs? There isn't a separate `MSCAFactory` right?

Reply:
There can be any number of factories for an MSCA which is why I added this `withFactory` concept. The other factory that exists already is the MultiOwnerWithTokenReceiver factory